### PR TITLE
Remove other event fields ignored by server

### PR DESCRIPTION
**Pull request by ashmrtn**
Update the set of ignored fields for event restores. Most important inclusion is the `iCalUId_v2` field which will cause failures if it's not removed.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/exchange/transform.go [hunk 1]
```diff
@@ -58,6 +58,28 @@ func toMessage(orig models.Messageable) models.Messageable {
...
+	// Appears to be duplicate of the iCalUId.
+	"uid",
+	// Navigation links from the calendar to the event itself.
+	"calendar@odata.associationLink",
+	"calendar@odata.navigationLink",
+	// Seems like info about the request that generated the data response.
+	"@odata.context",
+	// Appears to be similar to the change tag.
+	"@odata.etag",
+	// Remove exceptions for recurring events. These will be present in objects
```
Comment (pandeyabs) [inline src/internal/m365/collection/exchange/transform.go:77]: Should we create an issue to track this future change?

File: src/internal/m365/collection/exchange/transform.go [hunk 2]
```diff
@@ -58,6 +58,28 @@ func toMessage(orig models.Messageable) models.Messageable {
 	return CloneMessageableFields(orig, message)
 }
 
+// eventAdditionalUnneededFields contains a set of fields that don't have API
+// support right now but are ignored by or may cause errors in M365 when
+// restoring items.
+var eventAdditionalUnneededFields = []string{
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/exchange/transform.go:64]:
```suggestion
// eventUnsupportedAdditionalData lists the set of additionalData keys
// which are not needed for backup completion and may cause errors
// in Graph API when restoring items.
var eventUnsupportedAdditionalData = []string{
```

**Review by ryanfkeepers — APPROVED** [review]
lgtm.  Just a nit on naming.

File: src/internal/m365/collection/exchange/transform.go [hunk 3]
```diff
@@ -58,6 +58,28 @@ func toMessage(orig models.Messageable) models.Messageable {
...
+	// Will cause errors about needing to put extension data in their own requests
+	// if not removed.
+	"iCalUId_v2",
+	// Appears to be duplicate of the iCalUId.
+	"uid",
+	// Navigation links from the calendar to the event itself.
+	"calendar@odata.associationLink",
+	"calendar@odata.navigationLink",
```
Comment (pandeyabs) [inline src/internal/m365/collection/exchange/transform.go:72]: Curious, does graph auto populate any of these fields after we restore? i.e.if we fetch the event after restoring it, do these fields appear empty or filled? Wondering if removing `navigationLink` and such has any UX impact to the mailbox owner.

Comment (ashmrtn) [reply src/internal/m365/collection/exchange/transform.go:72]: yes, it auto populates them. The field itself contains the calendar ID and has the form
```
https://graph.microsoft.com/beta/users('<user_name>')/calendars('<calendar_id>')/$ref
```

File: src/internal/m365/collection/exchange/transform.go (hunk 1 above)
Comment (ashmrtn) [reply src/internal/m365/collection/exchange/transform.go:77]: I believe recurring events code is already in main

**Pull request merged**